solr/contrib/ltr/src/java/org/apache/solr/ltr/ranking/ModelQuery.java

```diff
@@ -152,12 +159,28 @@ public SolrQueryRequest getRequest() {
     return request;
   }
 
-  @Override  
-  public ModelWeight createWeight(IndexSearcher searcher, boolean needsScores, float boost) throws IOException {
+  @Override
+  public ModelWeight createWeight(IndexSearcher searcher, boolean needsScores, float boost)
+      throws IOException {   
+    long  start = System.currentTimeMillis();
+    
+    ModelWeight mw = null;
+    if(LTRThreadInterface.maxThreads <= 1){
```

Comment:
Not sure what is the right thing to do, here. On one hand, domains like news which have a small number of features can just set the number of threads to 1 in the config file to avoid parallelism. OTOH, if a domain like HL which mostly uses parallelism ends up using a small number of features for some queries, it might be unnecessary to use multiple threads. Leaving as it for now to keep it simple.